cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1479,9 +1534,9 @@ instance Error TxBodyError where
       "parameters to hash"
 
 
-makeTransactionBody :: forall era. ()
-  => IsCardanoEra era
-  => ScriptValidity -- ^ Mark script as expected to pass or fail validation
+makeTransactionBody :: forall era.
+     IsCardanoEra era
+  => TxBodyScriptValidity era -- ^ Mark script as expected to pass or fail validation
```

Comment:
So we shouldn't infect `makeTransactionBody` with a new parameter. We should add `TxBodyScriptValidity era` to `TxBodyContent` and feed the value in that way.

Reply:
This change is now in.  I'm liking this approach.